### DISCUSSION TITLE
Adding a Slave Zone

### DISCUSSION THREAD

**Discussion by lanhampr** [ngoduykhanh/PowerDNS-Admin#1409] [Q&A] [answered]
It doesn't seem to add slave zone's properly or we still have something jacked.  When I add a new slave zone, add the master IP the record the slave zone is not updated with such data and the slave server will not pull the record.  We have even tried making the slave server a superslave which I would expect the slave server always pull the zone from the master IP. We have not gotten the slave's to work unless we manually update the slave primary with the ip of the Master.

**Comment by AzorianMatt** [comment]
@jbe-dw @AgentTNT @corubba Are you one of you able to confidently answer this question by chance?

**Comment by metrax** [comment]
@lanhampr please login to your secondary and execute the follwing command:
`pdnsutil show-zone <zonename>`
Here you can see, if pda added the zone successfully, like here:
```
...
This is a Slave zone
Primaries: XXX.XXX.XXX.XXX:53
Last time we got update from primary: Mon 2023-02-27 21:09:26
...
```
If everthing is correct here, you have a problem with your powerdns servers. In this case, please check your logs.
If not, please let us know, than we will check this issue.

**Reply by lanhampr** [reply]
In the Master I added junkjunkjunk.com as a Primary domain.  I added same domain in Slave as Secondary and added IP of Master. Here are the results:

[root@ns-2 /]# pdnsutil show-zone junkjunkjunk.com
Feb 28 11:44:30 gmysql Connection successful. Connected to database 'pda' on 'db'.
Feb 28 11:44:30 gmysql Connection successful. Connected to database 'pda' on 'db'.
Feb 28 11:44:30 No serial for 'junkjunkjunk.com' found - zone is missing?
This zone is owned by clearnetworx
This is a Slave zone
Primaries: 
Last time we got update from primary: Never
No SOA serial found in database
Zone is not actively secured
Metadata items: 
	SOA-EDIT-API	DEFAULT
No keys for zone 'junkjunkjunk.com'.

The Slave server Zone Primary field is empty.

**Reply by metrax** [reply]
Which powerdns version are you using? It looks very old, because it using using "Slave" instead of "Secondary"

**Comment by lanhampr** [comment]
4.6.0

**Reply by metrax** [reply]
Mhhh, i thought, that the renaming was at 4.5.0. But i can check maybe later.
Please have a look to your PDA Logfiles, is it showing any error, when you are adding the slave zone?

**Comment by lanhampr** [comment]
It's a docker container using image: pschiffe/pdns-mysql:latest.  I am searching for log files within container.....

**Reply by metrax** [reply]
We need the PDA (PowerDNS-Admin) Logs, not the PDNS ;-)

**Comment by lanhampr** [comment]
Ok, I opened pschiffe/pdns-admin-uwsgi but I am unclear where the logs are stored.

**Reply by metrax** [reply]
At our official docker cointainer you can get it with docker logs <container>

Please understand that, we cannot support 3rd pardy docker images.
You can find the official image at:
https://hub.docker.com/r/powerdnsadmin/pda-legacy/tags/
I recommend to use version 0.3.0, due to "latest" is currently the development path.

**Comment by lanhampr** [comment]
I am assuming there may be various versions of ADMIN docker containers for PDNS.  Do you also have a recommendation for PDNS and PDNS Recursor containers?

**Reply by AzorianMatt** [reply]
@lanhampr there are a number of versions floating around out there, but they won't generally be kept up-to-date with the actual project repositories and releases so it's best you stick to the official container.

As far as containers for PowerDNS server binaries, it's hit or miss. I eventually hope to have my automated build strategy online so that every PDNS version is built automatically within a day of it's release. I do have some slightly older versions available here: https://hub.docker.com/u/azoriansolutions.

**Comment by lanhampr** [comment]
Thank you for your help.

**Comment by AzorianMatt** [comment, marked as answer]
As it was determined that an unofficial deployment was being used, I am considering this particular scenario as unsupported and thus, the potential issue cannot be verified as a PDA issue.
